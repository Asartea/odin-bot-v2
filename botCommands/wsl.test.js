const command = require('./wsl');

describe('!wsl', () => {
  describe('regex', () => {
    it.each([
      ['!wsl'],
      [' !wsl'],
      ['!wsl @odin-bot'],
      ['@odin-bot !wsl'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['wndows'],
      ['wsl'],
      ['window'],
      ['!wndows'],
      ['!window'],
      ['! wsl'],
      ['!awsl'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! !wsl'],
      ['Don\'t worry about !wsl'],
      ['Hey @odin-bot, !wsl'],
      ['!@odin-bot ^ !me !wsl !tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user!wsl'],
      ['it\'s about!wsl'],
      ['!wslanillusion'],
      ['!wsl!'],
      ['!wsl*'],
      ['!wsl...'],
    ])("'%s' - command should be its own word!group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
