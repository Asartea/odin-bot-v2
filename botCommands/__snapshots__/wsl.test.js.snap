// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!wsl callback returns correct output 1`] = `
{
  "embeds": [
    {
      "color": 13407555,
      "description": "For help with using WSL please use the WSL channel: <#1077759838785904640>",
      "title": "WSL",
    },
  ],
}
`;
